index.html

```diff
@@ -4292,6 +4306,493 @@ <h3>Proof Formats</h3>
+statements in <a href="#contexts"></a>. If not, return
+<var>dataModelVerified</var>, which will contain a false verification status.
+          </li>
+          <li>
+Starting at the top level, recursively process each object <var>obj</var> in
+<var>document</var>, setting <a href="#MALFORMED_VALUE_ERROR">
+MALFORMED_VALUE_ERROR</a> errors in
+<var>dataModelVerified</var>.<var>errors</var> if errors are found, by
+performing the following checks:
+            <ol class="algorithm">
+              <li>
+If present, ensure that the `id` property conforms to the "MUST" statements in
+Section <a href="#identifiers"></a>.
+              </li>
+              <li>
+If present, ensure that the `type` property conforms to the "MUST" statements in
+Section <a href="#types"></a>.
+              </li>
+              <li>
+If present, ensure that the `name` and/or `description` properties conform to
+the "MUST" statements in Section <a href="#names-and-descriptions"></a>.
+              </li>
+              <li>
+If present, ensure that the `proof` property conforms to
+the "MUST" statements in Section <a href="#securing-mechanisms"></a>.
+              </li>
+              <li>
+If the `type` property is present and contains the
+`VerifiablePresentation` type, ensure that the object conforms to the
+"MUST" statements in Section <a href="#presentations-0"></a>.
+              </li>
+              <li>
+If the `type` property is present and contains the
+`VerifiableCredential` type, perform the following checks:
+                <ol class="algorithm">
+                  <li>
+Ensure that the `credentialSubject` property conforms to the
+"MUST" statements in Section <a href="#credential-subject"></a>.
+                  </li>
+                  <li>
+Ensure that the `issuer` property conforms to the
+"MUST" statements in Section <a href="#issuer"></a>.
+                  </li>
+                  <li>
+If present, ensure that the `validFrom` and/or `validUntil` properties conform
+to the "MUST" statements in Section <a href="#validity-period"></a>.
+                  </li>
+                  <li>
+If present, ensure that the `credentialStatus` property conforms to
+the "MUST" statements in Section <a href="#status"></a>.
+                  </li>
+                  <li>
+If present, ensure that the `credentialSchema` property conforms to
+the "MUST" statements in Section <a href="#data-schemas"></a>.
+                  </li>
+                  <li>
+If present, ensure that the `relatedResource` property conforms to
+the "MUST" statements in Section <a href="#integrity-of-related-resources"></a>.
+                  </li>
+                  <li>
+If present, ensure that the `refreshService` property conforms to
+the "MUST" statements in Section <a href="#refreshing"></a>.
+                  </li>
+                  <li>
+If present, ensure that the `evidence` property conforms to
+the "MUST" statements in Section <a href="#evidence"></a>.
```

Comment:
All these MUST statements can be removed, by stating that the result of the "Verify Securing Mechanism" operation in either data integrity or vc-jose-cose or ... something else MUST be a conforming document.

Reply:
These are here because I was worried in https://github.com/w3c/vc-data-model/pull/1338#discussion_r1388509458 that it was hard to find all the things that a verifier needs to check to be sure something's a "conforming document". That is, @msporny started with text like you're suggesting, and I asked him to add more details, although not specifically these details. I suspect we can find a shared suggestion instead of making Manu mediate between us. :)

My basic worry is that the [definition of a "conforming document"](https://w3c.github.io/vc-data-model/#dfn-conforming-document) is written for the "authors" of those documents, usually issuers. Manu has made a lot of progress toward making that definition also useful for verifiers, but it's hard to maintain that state through spec evolution. Listing all the fields that need to be checked should help verifiers, because it gives them a narrower part of the document that they need to search for "MUST", but in my mind it's not as good as writing out the properties that need to be checked here.

Also note that listing the properties here makes it possible to omit "proof" from the list. If this just said verifiers need to check that the document is conforming, the "MUST"s in https://w3c.github.io/vc-data-model/#securing-verifiable-credentials would have to be included.

Reply:
As @jyasskin said, that's where I started @OR13, and this is where we needed to get to in order to address @jyasskin's concerns. If you want to make an attempt at a change, provide concrete change requests that @jyasskin (and the rest of the WG) can agree to.

Reply:
@OR13's suggestion of improving the definition of "conforming document" and using it to replace this expanded list of properties, sounds like a good way to go for at least this PR. I'm still not completely comfortable with the declarative style, but I don't see any uses of "conforming document" that would make me actually object to it.

Note that @OR13's definition as _"a W3C VCDM v2.0 conforming document, is a compact JSON-LD document, that follows all the MUST statements in the W3C VCDM v2.0 specification."_ is usefully different from the [current spec language](https://w3c.github.io/vc-data-model/#dfn-conforming-document) of _"any concrete expression of the data model that complies with the normative statements in this specification"_. In particular, the "normative statements" include the SHOULD-level statements that are hard for a verifier to check. [conforming verifier implementation](https://w3c.github.io/vc-data-model/#dfn-conforming-verifier-implementation) does better with "MUST check that each required property satisfies the normative requirements for that property", because arguably a SHOULD isn't a requirement.

Reply:
As a beginner verifier developer who's unfamiliar with all this, I would appreciate something like the above list. It might be better presented as a couple of tables, along the lines of —
```
<p>
Ensure that the following properties are present, and that they conform
to the "MUST" statements in the corresponding Section.
</p>
<table>
  <tr>
    <th>Property</th>
    <th>Section</th>
  </tr>
  <tr>
    <td>`credentialSubject`</td>
    <td><a href="#credential-subject"></a></td>
  </tr>
</table>
```
and
```
<p>
If any of the following properties are present, ensure they conform to
the "MUST" statements in the corresponding Section.
</p>
<table>
...
</table>
```